### PR TITLE
Entrypoints Hotfix

### PR DESCRIPTION
**Pull request by ZandroFargnoli**
Referenced in issue: 
https://github.com/ARM-software/vulkan_best_practice_for_mobile_developers/issues/28 

Entrypoint .exes can run without the same usage required by the `vulkan_best_practice.exe`

### CODE REVIEW

File: framework/platform/options.cpp [hunk 1]
```diff
@@ -24,24 +24,33 @@
...
+
 {
+	if (usage.size() != 0)
+	{
+		this->usage        = usage;
+		this->parse_result = docopt::docopt(usage, args, false);
+	}
+	else
+	{
+		LOGW("Application usage empty");
```
Comment (AttilioProvenzano-ARM) [inline framework/platform/options.cpp]: Would this warning be printed every time we run one of the sample entry points?

File: framework/platform/options.cpp [hunk 2]
```diff
@@ -24,24 +24,33 @@
...
 
 bool Options::contains(const std::string &argument) const
 {
-	assert(parse_result.count(argument) != 0 && "Argument not specified in usage");
-
-	if (auto result = parse_result.at(argument))
+	if (parse_result.count(argument) != 0)
 	{
-		if (result.isBool())
+		if (auto result = parse_result.at(argument))
```
Comment (AttilioProvenzano-ARM) [inline framework/platform/options.cpp]: This could be a `const auto&`

File: framework/platform/options.cpp (hunk 1 above)
Comment (ZandroFargnoli) [reply framework/platform/options.cpp]: Yes

File: framework/platform/options.cpp (hunk 2 above)
Comment (ZandroFargnoli) [reply framework/platform/options.cpp]: It can 👍

File: framework/platform/options.cpp (hunk 1 above)
Comment (AttilioProvenzano-ARM) [reply framework/platform/options.cpp]: I think we should remove the warning them, it's a normal use case to use the sample entry points. An empty usage can never happen when calling the main `vulkan_best_practice` app, if I'm not mistaken.

Comment (ZandroFargnoli) [reply framework/platform/options.cpp]: I had a warning for the meantime in the hotfix, as I was planning on changing the entire entry-point template for samples to be separate to Vulkan Best Practices so they don't even parse the arguments (i.e. you shouldn't be calling parse arguments if there is no usage, hence the warning).

I guess this also works though, I will remove it.

**Pull request merged**